csrc/optimization/alias_analysis.cpp

```diff
@@ -374,15 +374,15 @@ const TensorView* AliasAnalysisResult::getAliasedInput(
 void AliasAnalysisResult::finalize(Fusion* fusion) {
   for (TensorView* out :
        ir_utils::filterByType<TensorView>(fusion->outputs())) {
-    // Lazy move: we could check compatibility and only give up when
-    // the allocation domain is incompatible with what we prefer for
-    // aliasing.
-    if (out->hasAllocation()) {
+    const Val* in = findRoot(out);
+    if (!in->isFusionInput()) {
```

Comment:
what does this mean to output->output aliasing -> DuplicateOutputs.

We do have those in our cpp tests and I thought they are resolved via aliasing?!

Reply:
I'm stamping this, since if tests are passing this is likely just my misunderstanding of the code base. :laughing:

Reply:
Aren't they currently resolved by `outputs_map` you added? That code remains there in `csrc/executor.cpp` -- this PR doesn't touch it.

Reply:
sorry, I was dumb... yeah duplicated ouptuts isn't related here. I was thinking about common subexpression... but we don't have any support there neither at this point.